### ISSUE TITLE
Silent failure on `put`

### ISSUE THREAD

**Issue by Machiaweliczny** [redux-saga/redux-saga#856]
I'm using redux-saga@0.14.3 and encountered a weird silent error.

I'm trying to run saga per component I display on page. 

When I have 100 of them everything works fine, but when I have 200 of them it seems to fail on `put(channel, action)` or `put(action)` if communicating through the  store.

Code for saga looks like this:
```
function* mySaga(someChannel, id1, id2) {
  try {
    while(true) {
      const variable = yield select((state) => { return something; });
      console.log("before", id1);
      yield put(someChannel, someAction)
      console.log("after", id1);
      yield take(["string1", "string2"])
    }
  }
  catch(error){
    console.error("error in saga", error);
  } finally {
    console.info("finally executed")
  }
}
```

When I have 100 components everything works fine. I see `before` and `after` logged.

But when I have 200 components I see only `before` and no console logs from `catch` or `finally`.

As I don't have any error thrown at face it's hard for me to debug what might be wrong.

These sagas are spawned from other saga with usage of `yield spawn(mySaga, someChannel, id1, id2)` 

channel is created by `const someChannel = yield call(channel, buffers.expanding(200));` but as I mentioned before it doesn't matter if I use `put(channel, action)` or just `put(action)` the same thing happens.

**Comment by Machiaweliczny** [comment]
I tried even more simplified example to reduce the noise:
```
function* mySaga(someChannel, id1, id2) {
  try {
      console.log("before", id1);
      yield put(someChannel, someAction)
      console.log("after", id1);
  }
  catch(error){
    console.error("error in saga", error);
  } finally {
    console.info("finally executed")
  }
}
```
Same thing happens. With 200 it's hanging on `put`. With 100 everything is executed properly and finally is called.

**Comment by Andarist** [comment]
Could u provide a simple webpackbin with issue reproduced there?

**Comment by draganSm** [comment]
The given dummy code
```
    const N = 4000;
    for(var i = 0 ; i < N; ++i) {
      yield put({type: 'TEST' });
    }
```
results in `RangeError: Maximum call stack size exceeded` as `exec` and `flush` methods (scheduler.js) are calling each other.
Chrome: Version 56.0.2924.87 (64-bit)
```
{
  "name": "redux-saga",
  "version": "0.14.3",
  ...
```
Didn't have the issue on the latest FF even for N = 20000.
Hope that this is related to Machiaweliczny's issue.

**Comment by Andarist** [comment]
@draganSm 
I dont think thats connected, would be noticeable and the issue would include info about it. The problem u are referring too is being fixed though and gonna be merged in soon.

**Comment by wilsaj** [comment]
I started looking into this.

Here's a webpackbin that reproduces the issue: https://www.webpackbin.com/bins/-Ki6juosjlVKWfyw7-_O  
(note: it's better to use browser devtools than the webpackbin console log because it's so noisy)

Things I noticed:
- It does not seem to be a problem when using a branch with the scheduler fix for #888  (tested locally)
- The workaround suggested for #605 with adding a trivial noop `delay` seems to work, though I don't know why

**Comment by Andarist** [comment]
I've just released a new version with #888 fix, should we call this resolved and reopen if needed later? cc @wilsaj @Machiaweliczny

**Comment by wilsaj** [comment]
Looks like it's fixed for this specific test case at least.:
redux-saga 0.14.6(broken) : https://www.webpackbin.com/bins/-Ki6juosjlVKWfyw7-_O
redux-saga 0.14.7(fixed) : https://www.webpackbin.com/bins/-KiB89zk_PXidxlFEIfg
